enricher/fabric8/src/main/java/io/fabric8/maven/enricher/fabric8/DockerHealthCheckEnricher.java

```diff
@@ -0,0 +1,124 @@
+ * This enricher could need a change when Dockerfile health checks will be supported natively in Openshift or Kubernetes.
+ */
+public class DockerHealthCheckEnricher extends AbstractHealthCheckEnricher {
+
+    public DockerHealthCheckEnricher(EnricherContext buildContext) {
+        super(buildContext, "docker-health-check");
+    }
+
+    /**
+     * Try to provide a health check for any image defined in the config.
+     */
+    @Override
+    protected List<Container> getCandidateContainers(Deployment deployment) {
+        if (deployment.getSpec() != null &&
+                deployment.getSpec().getTemplate() != null &&
+                deployment.getSpec().getTemplate().getSpec() != null) {
+
+            List<Container> containers = deployment.getSpec().getTemplate().getSpec().getContainers();
+            if (containers != null) {
+                return containers;
+            }
+        }
+
+        return Collections.emptyList();
+    }
+
+    @Override
+    protected Probe getReadinessProbe(Container container) {
+        return getProbe(container);
+    }
+
+    @Override
+    protected Probe getLivenessProbe(Container container) {
+        return getProbe(container);
+    }
+
+    protected Probe getProbe(ImageConfiguration image) {
+        if (hasHealthCheck(image)) {
+            HealthCheckConfiguration health = image.getBuildConfiguration().getHealthCheck();
+            return new ProbeBuilder()
+                    .withExec(new ExecAction(health.getCmd().asStrings()))
+                    .withTimeoutSeconds(durationSeconds(health.getTimeout()))
+                    .withPeriodSeconds(durationSeconds(health.getInterval()))
+                    .withFailureThreshold(health.getRetries())
+                    .build();
+        }
+
+        return null;
+    }
+
+    protected Probe getProbe(Container container) {
+        String imageName = container.getImage();
+        if (imageName == null) {
+            return null;
+        }
+
+        ImageConfiguration containerImage = null;
+        List<ImageConfiguration> images = getImages();
+        if (images != null) {
+            List<ImageConfiguration> candidate = new LinkedList<>();
+            for (ImageConfiguration image : images) {
+                if (imageName.equals(image.getName())) {
+                    candidate.add(image);
+                }
+            }
+
+            if (candidate.size() == 1) {
+                // Just one match, choosing it
+                containerImage = candidate.get(0);
+            } else if (candidate.size() > 1) {
+                // More than one match, matching with alias
+                for (ImageConfiguration image : images) {
+                    String cName = KubernetesResourceUtil.extractContainerName(getProject(), image);
+                    if (cName != null && cName.equals(container.getName())) {
```

Comment:
If they container names match by alias, then of course they images must match, too (and hence are covered in the loop above). Also, I think its dangerous since the container name could be set directly in the image configuration and needs not be the default as it comes out from the utility methods. 

I would skip this step here.